From 585a7f02adad8a1b3135d9ae80dfe62f38a73b4e Mon Sep 17 00:00:00 2001
From: Colin McAllister <colinmca242@gmail.com>
Date: Fri, 13 Feb 2026 11:17:53 -0600
Subject: [PATCH 2/8] gh-144787: [tests] Allow TLS v1.2 to be minimum version
 (GH-144790)

Allow TLS v1.2 to be minimum version

Updates test_min_max_version to allow TLS v1.2 to be minimum version if
TLS 1.0 and 1.1 are disabled in OpenSSL.

(cherry picked from commit d625f7da33bf8eb57fb7e1a05deae3f68bf4d00f)
---
 Lib/test/test_ssl.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Lib/test/test_ssl.py b/Lib/test/test_ssl.py
index aa8ce81db66..e36bb43b64b 100644
--- a/Lib/test/test_ssl.py
+++ b/Lib/test/test_ssl.py
@@ -1330,7 +1330,12 @@ def test_min_max_version(self):
         ctx.maximum_version = ssl.TLSVersion.MINIMUM_SUPPORTED
         self.assertIn(
             ctx.maximum_version,
-            {ssl.TLSVersion.TLSv1, ssl.TLSVersion.TLSv1_1, ssl.TLSVersion.SSLv3}
+            {
+                ssl.TLSVersion.TLSv1,
+                ssl.TLSVersion.TLSv1_1,
+                ssl.TLSVersion.TLSv1_2,
+                ssl.TLSVersion.SSLv3,
+            }
         )
 
         ctx.minimum_version = ssl.TLSVersion.MAXIMUM_SUPPORTED
-- 
2.53.0

